applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -351,6 +424,10 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
             //    grow = false;
             //}
 
+            if (mStoreNonconvergedSolutionsFlag) {
+                CollectCurrentSolution();
+            }
```

Comment:
Just a question to verify my understanding: since the boolean is called `mStoreNonconvergedSolutionsFlag`, should it also be checked if the solution was converged and only collect the current solution with it hasn't? If we only want to save non-converged solutions, I'd propose the following:
```suggestion
            if (mStoreNonconvergedSolutionsFlag && !converged) {
                CollectCurrentSolution();
            }
```

or put it as the `else` after the if-statement of line 431